### PR TITLE
fetch all available concepts from webconcepts.info

### LINKED ISSUE #1: load concept info dynamically
http://webconcepts.info/concepts.json has all info on all concepts, including their names (in english only so far) and all values. maybe that's a bit more robust than hardcoding the ones that you currently have? if you need more info about the JSON, or maybe some additional info, feel free to raise issues over at https://github.com/dret/webconcepts!

### PR DESCRIPTION
**Pull request by tpluscode**
fixes #1

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
-  'method': uri('http-method'),
-  'relation': uri('link-relation')
-}
+var concepts = {};
+var conceptList = [];
 
-var conceptList = Object.keys(concepts).join(', ')
+request(allConcepts, (error, response, body) => {
+  if (error || response.statusCode >= 500){
+    return;
```
Comment (tpluscode) [inline index.js]: I'm not sure what can be done about a failed initialization. Maybe it should be retried whenever the `/web-concept` command is used if not initialized properly at start?

Comment (pmhsfelix) [reply index.js]: This will be done on *every* request, right? I'm reluctant on paying that extra price for now.

**Review by pmhsfelix — COMMENTED** [review]
Perhaps use [https://webtask.io/docs/storage] as a cache for the full concept list.

File: index.js (hunk 1 above)
Comment (tpluscode) [reply index.js]: What will you pay extra for. Each outgoing request, processing time or bandwidth?

Comment (pmhsfelix) [reply index.js]: By "pay" I meant the time required for that extra call. Now there will be two outbound calsl for one inbound call from Slack, which increases the latency for the response.

Comment (tpluscode) [reply index.js]: Oh, ok. I assumed that only the exported module is executed at each request. I now realize that webtask.io is a serverless service. I'll look into that storage feature

**Comment by tpluscode** [comment]
I figured that given that all concepts' information is already present in the `concepts.json` document, it's only necessary to make one call after all.

It could be cached regardless, but the webtask.io storage doesn't look like a great choice because it has no expiration policy. 

Also, I made the lookup case-insensitive

**Comment by pmhsfelix** [comment]
I'll try to take a look later today. Regarding the expiration policy, we can add a time stamp to the store information and invalidate it explicitly when it expires. Thanks!.

**Comment by tpluscode** [comment]
With the lack of other options a timestamp would work :)

**Comment by dret** [comment]
i hope this week's JSON changes did not break your code? i have added a bit of documentation to the site to explain the current JSON structure: http://webconcepts.info/JSON

**Comment by dret** [comment]
and as mentioned by @tpluscode and acknowledged in https://github.com/dret/webconcepts/issues/28, there currently is this oddity that concepts can have multiple definitions, and there's not even any indication what the "right" one is. this bugs me, but i am not certain how to best resolve this. any ideas would be very welcome!

**Comment by tpluscode** [comment]
@dret when did you apply those changes? when I posted this PR the code seemingly worked fine

**Comment by pmhsfelix** [comment]
I've already merged and successfully tried out @tpluscode changes. The new version is already being used on https://httpapis.slack.com/

**Comment by dret** [comment]
On 2016-12-17 12:17, Tomasz Pluskiewicz wrote:
> @dret <https://github.com/dret> when did you apply those changes? when I
> posted this PR the code seemingly worked fine

https://github.com/dret/webconcepts/commit/14901a2591b382c57fc67550457adacd74576707 
was the big one, 2 days ago.

**Comment by dret** [comment]
On 2016-12-17 12:19, Pedro Felix wrote:
> I've already merged and successfully tried out @tpluscode
> <https://github.com/tpluscode> changes. The new version is already being
> used on https://httpapis.slack.com/

maybe nothing changed that actually affected your code. the concept URIs 
changed, but that shouldn't change things unless you had that pattern 
hardcoded somewhere.

i added a link to a registry (if there is one), but that's something you 
simply might want to use in future versions:

https://github.com/dret/webconcepts/commit/eaf85b6ee03fab8e79e93119a40c1f24e44a6fd6

**Comment by tpluscode** [comment]
there are no hardcoded URIs now. all is taken from the `concepts.json` resource

**Comment by dret** [comment]
On 2016-12-17 12:34, Tomasz Pluskiewicz wrote:
> there are no hardcoded URIs now. all is taken from the |concepts.json|
> resource

all hail hypermedia! ;-)

**Pull request merged**